lib/use_packs/private.rb

```diff
@@ -358,8 +359,8 @@ def self.add_readme_todo(package)
       ).returns(ParsePackwerk::Package)
     end
     def self.create_pack_if_not_exists!(pack_name:, enforce_privacy:, enforce_dependencies:, team: nil)
-      if PERMITTED_PACK_LOCATIONS.none? { |permitted_location| pack_name.start_with?(permitted_location) }
-        raise StandardError, "UsePacks only supports packages in the the following directories: #{PERMITTED_PACK_LOCATIONS.inspect}. Please make sure to pass in the name of the pack including the full directory path, e.g. `packs/my_pack`."
+      if PERMITTED_PACK_LOCATIONS.none? { |permitted_location| pack_name.match?(permitted_location) }
```

Comment:
I'm not sure this will work. Here's an example:
```
Zenpayroll DEVELOPMENT (main) 6> 'foo/bar/**'.scan(/\w+/)
=> ["foo", "bar"]
Zenpayroll DEVELOPMENT (main) 7> 'bar/blah'.match?('bar')
=> true
```

This has a false positive for a pack name in `bar` even though it's not one of the permitted pack locations.

Could you write a test for this that fails with the current implementation and then we can find another implementation that passes all tests.